co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstance.java

```diff
@@ -167,27 +234,40 @@ public void bind() throws IOException {
      * @throws IOException If there is an issue with the underlying socket object or
      *                     methods
      */
-    public void sendMsgs(byte[] data) throws IOException {
-        if (rxMsgsSocket == null) {
-            throw new IllegalStateException("Attempted to send data before opening socket");
-        }
-        DatagramPacket packet = new DatagramPacket(data, data.length, targetAddress, rxMessagePort);
-        rxMsgsSocket.send(packet);
-
+    public void sendV2xMsg(byte[] data) throws IOException {
+        sendPacket(data, rxMessagePort);
     }
 
     /**
-     * Sends time sync data to the Infrastructure Device communications interface configured at construction time.
+     * Sends time sync data to the Infrastrucutre Instance Time Sync port.
+     * 
+     * @param data The binary data to transmit
+     * @throws IOException If there is an issue with the underlying socket object or methods
+     */
+    public void sendTimeSyncMsg(byte[] data) throws IOException {
+        sendPacket(data, timeSyncPort);
+    }
+    /**
+     * Sends time sync data to the Infrastrucutre Instance Simulated Interaction port.
+     * 
      * @param data The binary data to transmit
      * @throws IOException If there is an issue with the underlying socket object or methods
      */
-    public void sendTimeSyncMsgs(byte[] data) throws IOException {
-        if (rxMsgsSocket == null) {
+    public void sendInteraction(byte[] data) throws IOException {
+        sendPacket(data, simulatedInteractionPort);
+    }
+    /**
+     * Method to send byte data to specified port using the infrastructure instance Datagramsocket.
+     * 
+     * @param data byte data to send using Datagramsocket.
+     * @param port in integer format to send Datagram to.
+     * @throws IOException
+     */
+    public void sendPacket(byte[] data, int port) throws IOException {
```

Comment:
New method created to avoid code duplication between sendTimeSyncMsg sendV2xMsg and sendInteraction